Issuer plugin for Google CA Manager

odopertchouk opened the pull request:
A plugin that handles certificate issuance via Google CA Manager.

The plugin requires `GOOGLE_ACCOUNT_CREDENTIALS` config variable, which should point at the file containing credentials that Lemur is using to connect to Google Cloud Platform. 
These credentials normally would be for a service account that has permissions `privateca.certificates.update`, `privateca.certificates.create` and `privateca.certificateAuthorities.get`

This requires packages
- google-cloud-private-ca package
- protobuf
- google-cloud-private-ca

jmcrawford45 commented on lemur/plugins/lemur_google_ca/plugin.py:
```diff
@@ -0,0 +1,214 @@
+
+
+def fetch_authority(ca_path: str) -> tuple[str, str]:
+    client = privateca_v1.CertificateAuthorityServiceClient()
```
Does this need application credentials as well?

odopertchouk replied:
Good catch!

jmcrawford45 reviewed (commented):
First of all, thank you for the work you've put into creating this new issuer plugin for the Google Certificate Authority Service. I've left a few minor comments in the code for things that could be improved. Before merging, we'd like to have some tests in place that cover the main functionality of your plugin, such as certificate creation, revocation, and fetching. Additionally, if you could add a changelog entry to make consumers aware of the new functionality, that'd be appreciated.

We appreciate your contribution and look forward to seeing the improvements and tests. If you need any help or clarification, feel free to ask.

odopertchouk commented:
> First of all, thank you for the work you've put into creating this new issuer plugin for the Google Certificate Authority Service. I've left a few minor comments in the code for things that could be improved. Before merging, we'd like to have some tests in place that cover the main functionality of your plugin, such as certificate creation, revocation, and fetching. Additionally, if you could add a changelog entry to make consumers aware of the new functionality, that'd be appreciated.
> 
> We appreciate your contribution and look forward to seeing the improvements and tests. If you need any help or clarification, feel free to ask.

I have added some tests, but can you check that the tests are consistent enough with the rest of the project? I have tried to follow examples from other plugins, but not everything carries over.

Similarly, I have added a changelog entry under 1.7.0 - `2024-01-17`  but I am not sure if it's the right location.

odopertchouk commented:
@jmcrawford45 The build checks have been stuck all day, is there something wrong, or they just need more time?

odopertchouk commented:
@jmcrawford45  now the build fails with an error
`⚠️ Coverage reporter does not yet know how to process this file: .coverage`  
Not sure how to proceed, as I have no experience with it

jmcrawford45 commented:
> @jmcrawford45 now the build fails with an error `⚠️ Coverage reporter does not yet know how to process this file: .coverage` Not sure how to proceed, as I have no experience with it

This is a known upstream issue https://github.com/coverallsapp/github-action/issues/205 I've temporarily disabled coveralls checks and will get your PR merged in today. Thanks again for your contribution and patience!
